biorxiv/corpora_comparison/biorxiv_pmc_figure_generator.py

```diff
@@ -0,0 +1,165 @@
+    full_text_comparison
+    .assign(repository=lambda x: x.odds_ratio.apply(lambda ratio: "bioRxiv" if ratio > 1 else "PMC"))
+    .sort_values("odds_ratio", ascending=False)
+)
+full_text_comparison
+
+
+# In[4]:
+
+
+subset = 20
+plot_df = (
+    full_text_comparison.head(subset).append(full_text_comparison.tail(subset))
+    .assign(odds_ratio=lambda x: x.odds_ratio.apply(lambda ratio: ratio if ratio > 1 else 1/ratio))
+    .assign(label=lambda x: x.apply(
+        lambda row: (
+            f"<b>{row.lemma}</b><br />bioRxiv:{row.biorxiv_count:,}"
+            f"<br />PMC:{row.pmc_count:,}<br />"
+            f"Odds Ratio:{row.odds_ratio:.2f}"
+        ),
+        axis=1
+    ))
+    
+)
+plot_df.head()
+
+
+# In[5]:
+
+
+fig=go.Figure(
+    go.Treemap(
+        branchvalues="remainder",
+        labels = ["bioRxiv","PMC"] + plot_df.label.tolist(),
+        parents = ["", ""] + plot_df.repository.tolist(),
+        values = [0,0] + plot_df.odds_ratio.tolist(),
+        textinfo = "label",
+        textfont = {'size':17}
+    )
+)
+fig.update_layout(
+    width=1280, 
+    height=720,
+    treemapcolorway=["#a6cee3", "#b2df8a"],
+    title={
+        'text':f"Comparison of Top {plot_df.shape[0]} Words between bioRxiv and Pubmed Central (PMC)",
+        'y':0.9,
+        'x':0.5,
+        'xanchor': 'center',
+        'yanchor': 'top'}
+)
+fig.update_layout(uniformtext=dict(minsize=10))
+fig.show()
+
+
+# In[6]:
+
+
+fig.write_image("output/figures/biorxiv_vs_pubmed_central.svg", scale=2.5)
+fig.write_image("output/figures/biorxiv_vs_pubmed_central.png", scale=2.5)
+
+
+# # Preprint vs Published
+
+# In[7]:
+
+
+preprint_published_comparison = pd.read_csv("output/preprint_to_published_comparison.tsv", sep="\t")
+preprint_published_comparison.head()
+
+
+# In[8]:
+
+
+preprint_published_comparison = (
+    preprint_published_comparison
+    .assign(repository=lambda x: x.odds_ratio.apply(lambda ratio: "preprint" if ratio > 1 else "published"))
+    .sort_values("odds_ratio", ascending=False)
+)
+preprint_published_comparison
+
+
+# In[9]:
+
+
+subset = 20
+plot_df = (
+    preprint_published_comparison.head(subset).append(preprint_published_comparison.tail(subset))
+    .assign(odds_ratio=lambda x: x.odds_ratio.apply(lambda ratio: ratio if ratio > 1 else 1/ratio))
+    .assign(label=lambda x: x.apply(
+        lambda row: (
+            f"<b>{row.lemma}</b><br />Preprint:{row.preprint_count:,}"
+            f"<br />Published:{row.published_count:,}<br />"
+            f"Odds Ratio:{row.odds_ratio:.2f}"
+        ),
+        axis=1
+    ))
+    
+)
+plot_df.head()
+
+
+# In[10]:
+
+
+fig=go.Figure(
+    go.Treemap(
+        branchvalues="remainder",
+        labels = ["preprint","published"] + plot_df.label.tolist(),
+        parents = ["", ""] + plot_df.repository.tolist(),
+        values = [0,0] + plot_df.odds_ratio.tolist(),
+        textinfo = "label",
+        textfont = {
+            'size':15
+        }
+    )
+)
+fig.update_layout(
```

Comment:
I'm not sure the best way to visualize this information. My instinct would just be to have two ranked lists of words because it would be straightforward to compare. 

The only issues I had with this visualization is that the smaller words I ended up ignoring when i looked at it just because their font was so small, though i understand this was to show that it was less specific. I think I personally had a little trouble going back and forth between the two plots to compare the words and the words were scattered around so I would loose my place a little when I'd glace back.

In short I'm not sure of another option other than ranked lists right now and i I think the way you have it is a good start.

Reply:
Which figure did you see? Was it the blue and green treemap? If so that is the wrong figure. You were suppose to see a point plot with a barchart. I updated the graph. Please take a look at this one.

Reply:
Yup looked at the wrong plot!

These bar plots look a lot better in terms of comparing the data. Its interesting that the terms with odd ratio >1 reflect very slight increases in the preprints, but for terms with odd ratio <1 the difference in counts is much larger in the published papers.  Not sure if it makes more sense to sort by the change in reference to 1 for this